src/libexpr/attr-set.hh

```diff
@@ -16,10 +16,10 @@ struct Value;
 struct Attr
 {
     SymbolPtr name;
-    Value * value;
     PosPtr pos;
+    Value * value;
```

Comment:
Maybe add a comment that the order of these was choosen purposely?

Reply:
added a comment describing the reasoning here. not sure an assertion has much value though, do you have a scenario in mind that might interfere here?

Reply:
Since it has a signitifcant impact I think the trivial issue could be that someone adds another field here without fully understanding the consequences. Also just some refactoring (prior to your comment) might have lead to someone moving them around (for various reasons) without knowing that the layout was chosen deliberately.

Reply:
added an assertion for the expected size with a warning about why it's important.